java-api/src/main/java/com/sap/cloud/security/token/SecurityContext.java

```diff
@@ -20,6 +20,41 @@ private SecurityContext() {
 	}
 
 	private static final ThreadLocal<Token> tokenStorage = new ThreadLocal<>();
+	private static final ThreadLocal<String> certificateStorage = new ThreadLocal<>();
```

Comment:
in the go client lib I've decided to store not the string but the x509 certificate object. Otherwise the different X509 validators needs to the same multiple ways as all of them require the object and not the string. is that possible in your implementation?

Reply:
If for client certificate validation in certificate comparison only DN needs to match, it makes sense to switch, but if the whole certificate needs to be matched then it's better to store base64 encoded cert.